modules/gdscript/gdscript_function.cpp

```diff
@@ -1583,15 +1583,26 @@ Variant GDScriptFunction::call(GDScriptInstance *p_instance, const Variant **p_a
 		GDScriptLanguage::get_singleton()->script_frame_time += time_taken - function_call_time;
 	}
 
-	if (ScriptDebugger::get_singleton())
-		GDScriptLanguage::get_singleton()->exit_function();
+	bool yielded = retvalue.is_ref() && Object::cast_to<GDScriptFunctionState>(retvalue);
```

Comment:
I'm not sure that's the best way to check if the function yielded. This might break in cases such as:
```gdscript
func a():
  yield()
func b():
  var x = a()
  # x.resume()
  return x # Not yielding, yet returning a GDScriptFunctionState
```

Reply:
> I guess we can find a way to check if the state correspond to the same function. Not sure how to do this.

Still gameable:
```gdscript
func a():
  var x = b()
  return b(x)
func b(z=null):
  if z != null:
    return z
  else:
    yield()
```

Reply:
So is any further change needed?

Reply:
Ping.

Reply:
Guess it could be fine as-is.